src/test/coll.rs

```diff
@@ -787,3 +788,43 @@ async fn watch_test_coll() {
         .build();
     coll.watch(pipeline, Some(options)).await.unwrap();
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn watch_options_test() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let client = EventClient::new().await;
+    if client.is_standalone() {
+        return;
+    }
+
+    let coll = client
+        .init_db_and_coll(function_name!(), function_name!())
+        .await;
+
+    let coll_options = CollectionOptions::builder()
+        .selection_criteria(SelectionCriteria::ReadPreference(ReadPreference::Primary))
+        .read_concern(ReadConcern::from(ReadConcernLevel::Available))
+        .build();
+
+    client
```

Comment:
the result of this line is unused, so the options set on it will not be included in the final change stream, causing this test to fail. Also, we need to create the collection in this test like we do in the other change stream tests.

I would encourage you to test these locally first, since waiting for evergreen patches can take some time, and doing so also saves on round trips of code review.

Reply:
fixed